https://github.com/MisterTea/EternalTerminal/commit/fde8a7cb084bb6bc005b8205f4257e7f65978521

Only the diff for src/base/TunnelUtils.hpp is used here, the rest is
retained as is.

From fde8a7cb084bb6bc005b8205f4257e7f65978521 Mon Sep 17 00:00:00 2001
From: kruceter <kruceter@proton.me>
Date: Thu, 26 Jan 2023 20:38:12 +0300
Subject: [PATCH] Fix cross build with GCC 12 (#564)

* src/base/TunnelUtils.hpp: replace "ETerminal.pb.h" with "Headers.hpp"

Fixes "strings" and "vectors" not having a type.

Headers.hpp already includes ETerminal.pb.h.

* external/cxxopts: update submodule

Needed in order to fix cross build with GCC 12.

Note that the submodule is updated to revision
c74846a891b3cc3bfa992d588b1295f528d43039 which is equal to the tagged
version 3.0.0.

This decision was made due to the changes in master of cxxopts breaking
EternalTerminal (such as removing cxxopts::OptionException from the
code).
---
 external/cxxopts         | 2 +-
 src/base/TunnelUtils.hpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/base/TunnelUtils.hpp b/src/base/TunnelUtils.hpp
index 207969ee..42e96cc4 100644
--- a/src/base/TunnelUtils.hpp
+++ b/src/base/TunnelUtils.hpp
@@ -1,7 +1,7 @@
 #ifndef __ET_TUNNEL_UTILS__
 #define __ET_TUNNEL_UTILS__
 
-#include "ETerminal.pb.h"
+#include "Headers.hpp"
 
 namespace et {
 
